### PR TITLE
Price Field Buggy Behaviour

### PR DESCRIPTION
**Pull request by chriistine**
## Notion ticket link

[Price Field Buggy Behaviour](https://www.notion.so/uwblueprintexecs/16fb020b93f641c8b384f76a1b3c769e?v=aedc8d99cd8b437cb556ece10f3c0e3c&p=129bce0d3b3f4ee6a342cf02f40dcc05)

## Implementation description
* Changed price type from string to number type and updated interface checker file in the backend
* Changed price type to string in the frontend by using AddStringInput instead of AddNumberInput - sent as a number type to the backend
* Added isValidPrice to ensure the validity of the price string  
* Ensured that price is always rounded to 2 decimal digits

## Steps to test
1.

## What should reviewers focus on?
* 

## Checklist
- [X] My PR name is descriptive and in imperative tense
- [X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [X] I have run the appropriate linter(s)
- [X] I have run docker-compose up and my project compiled
- [X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/CreateReview/BookModal.tsx [hunk 1]
```diff
@@ -139,7 +139,7 @@ const BookModal = (props: BookModalProps): React.ReactElement => {
 
       const bookFormat = book.formats[0];
       setFormat(bookFormat.format);
-      setPrice(parseInt(bookFormat.price, 10));
+      setPrice(bookFormat.price.toFixed(2));
```
Comment (jessiiepeng) [inline frontend/src/components/pages/CreateReview/BookModal.tsx:142]: This is throwing an error when I try viewing existing reviews.

**Review by jessiiepeng — COMMENTED** [review]
Everything works on my end except viewing pre-existing reviews and books.

**Review by Saber74 — CHANGES_REQUESTED** [review]
LGTM, saw the same issue as Jessie which would require additional work

**Comment by Saber74** [comment]
Okay so 1 additional comment is that even though you do verify if the price is valid, the user is still able to type in invalid prices and have it show on the screen. I suggest you rebase off master and use the new functionality that I created for the form control where you can add a regex pattern, such that if the regex pattern is not met, the field is not updated as such the users can't type in invalid values. https://github.com/uwblueprint/ccbc/pull/105/files#diff-5fdd6052491714d922fb07ef7c78367df5818a9bb4f23b8cdbdd7aed3a0b418cR344

**Comment by chriistine** [comment]
as per discussion - migration will not be written for this ticket - will update db as necessary

**Review by Saber74 — APPROVED** [review]
Great work!! Can't type in invalid values now, and upon edit I can see 2 decimal places even if omitted initially.

**Pull request merged**